src/python_be.cc

```diff
@@ -1901,8 +1901,8 @@ TRITONBACKEND_Initialize(TRITONBACKEND_Backend* backend)
 
   std::unique_ptr<BackendState> backend_state(new BackendState());
   triton::common::TritonJson::Value cmdline;
-  backend_state->shm_default_byte_size = 64 * 1024 * 1024;  // 64 MBs
-  backend_state->shm_growth_byte_size = 64 * 1024 * 1024;   // 64 MBs
+  backend_state->shm_default_byte_size = 1 * 1024 * 1024;  // 1 MBs
+  backend_state->shm_growth_byte_size = 1 * 1024 * 1024;   // 1 MBs
```

Comment:
Out of curiosity, where did the 64MB minimum come from? And how early would the backend/model identify it needs to grow the size? Are there cases where it realizes it needs to grow by say 100MB, but will still only attempt to grow 1MB at a time for 100 times?

Reply:
> Out of curiosity, where did the 64MB minimum come from?

I believe it was originally suggested by one of the former team members.

> How early would the backend/model identify it needs to grow the size?

It would grow the region whenever it runs out of shared memory. https://github.com/triton-inference-server/python_backend/blob/0f2ce85f789dddffec51aba6c483f759675f66f1/src/shm_manager.h#L96-L99

> Are there cases where it realizes it needs to grow by say 100MB, but will still only attempt to grow 1MB at a time for 100 times?

There might be cases like this but I think it won't be common. If the user has a large 100MB tensor the region would grow to 100MB in one pass. I think the only case that this might affect the perf is if they gradually in small increments increase the shared memory usage. For those cases, the options are still exposed if they want to change the default.

> What is the [default constructed SHM Manager](https://github.com/triton-inference-server/python_backend/blob/0f2ce85f789dddffec51aba6c483f759675f66f1/src/shm_manager.cc#L97-L102) with default growth size of 1 KB used for? It looks like it's used for SHM Monitor testing, but can you summarize the reasoning / why different growth value?

I think this is only used in the shared memory monitor: https://github.com/triton-inference-server/python_backend/blob/0f2ce85f789dddffec51aba6c483f759675f66f1/src/shm_monitor/shm_monitor.cc so it is not used in the production Python backend code.